extension/js/common/api/email-provider/gmail/gmail.ts

```diff
@@ -277,14 +278,20 @@ export class Gmail extends EmailProviderApi implements EmailProviderInterface {
     if (format === 'full') {
       const bodies = GmailParser.findBodies(gmailMsg);
       const attachments = GmailParser.findAttachments(gmailMsg);
-      const fromTextBody = PgpArmor.clip(Buf.fromBase64UrlStr(bodies['text/plain'] || '').toUtfStr());
+      const textBody = Buf.fromBase64UrlStr(bodies['text/plain'] || '').toUtfStr();
+      const htmlBody = Xss.htmlSanitizeAndStripAllTags(Buf.fromBase64UrlStr(bodies['text/html'] || '').toUtfStr(), '\n');
+      const fromTextBody = PgpArmor.clip(textBody);
       if (fromTextBody) {
         return { armored: fromTextBody, subject, isPwdMsg };
       }
-      const fromHtmlBody = PgpArmor.clip(Xss.htmlSanitizeAndStripAllTags(Buf.fromBase64UrlStr(bodies['text/html'] || '').toUtfStr(), '\n'));
+      const fromHtmlBody = PgpArmor.clip(htmlBody);
       if (fromHtmlBody) {
         return { armored: fromHtmlBody, subject, isPwdMsg };
       }
+      if ((fromTextBody === undefined || fromHtmlBody === undefined) && attachments.length === 0) {
+        const clippedText = (textBody || htmlBody).substring((textBody || htmlBody).search('-----BEGIN PGP MESSAGE-----'));
```

Comment:
perhaps, `indexOf` is better than `search` since we're not dealing with regexes.
Also, instead of copying the string literal, I'd suggest to reference it via
`PgpArmor.headers('encryptedMsg').begin`

Reply:
thanks, I applied these changes now.